### PR TITLE
Fix allowed dtypes for mem_eff attention

### PR DESCRIPTION
**Pull request by drisspg**
# Summary

Fix issue bug in detecting mem eff capability for cuda devices less than sm80:
https://github.com/pytorch-labs/gpt-fast/issues/49

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### CODE REVIEW

File: test/test_transformers.py [hunk 1]
```diff
@@ -1503,8 +1504,9 @@ def test_nested_fails_on_padding_head_dim(self, device):
 
 
     @onlyCUDA
-    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_ATTENTION or not isSM5xDevice, "Does not support fused SDPA or not SM50 hardware")
-    def test_mem_efficient_fail_bfloat16_sm50(self, device):
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_ATTENTION or not isLessThanSM80Device,
+                     "Does not support fused SDPA or not less than sm80 hardware")
```
Comment (janeyx99) [inline test/test_transformers.py:1508]: This is a hard to parse comment, maybe more: 
"Current platform does not support fused SDPA or is an SM80+ device."

**Review by janeyx99 — APPROVED** [review]
Approving as this seems like a better state of world. However, are there tests verifying that halfs and floats _do_ work with SM80 below architectures?

**Comment by drisspg** [comment]
@janeyx99 Yup the test test_mem_efficient_attention_vs_math_ref_grads checks float16, float32 types on less than sm80. As well I left breadcrumbs for myself here since I only enabled bfloat16 greater than sm80.. I don't know how I missed this for so long..

**Comment by drisspg** [comment]
@jeanschmidt can you point me to the failing diff id I can find this info from: https://www.internalfb.com/intern/opensource/github/pull_request/689041750001160/

**Comment by jeanschmidt** [comment]
@drisspg sure, https://fburl.com/diff/nica74p1
